### PR TITLE
ONNX Export Support for Celu

### PR DESCRIPTION
**Pull request by KsenijaS**
Add ONNX export support for torch.nn.CELU

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -3138,6 +3122,35 @@ def forward(self, x):
         x = torch.randn(2, 5)
         self.run_test(ShapeModule(), (x,), rtol=1e-3, atol=1e-5)
 
+    @skipIfUnsupportedMinOpsetVersion(12)
+    def test_celu(self):
+        class Celu(torch.nn.Module):
+            def __init__(self):
+                super(Celu, self).__init__()
+                self.celu = torch.nn.CELU(alpha=1.0)
```
Comment (lara-hdr) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:3130]: add one test case with a default alpha and one test case with an alpha with a different value than the default value (which is 1) ?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -2967,22 +2967,6 @@ def forward(self, input, target):
 
         self.run_test(CrossEntropyLossMeanWeightIgnoreIndex(), input=(x, y))
 
-    def test_empty_branch(self):
```
Comment (lara-hdr) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:2970]: is there a reason this test was moved?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -3138,6 +3122,35 @@ def forward(self, x):
         x = torch.randn(2, 5)
         self.run_test(ShapeModule(), (x,), rtol=1e-3, atol=1e-5)
 
+    @skipIfUnsupportedMinOpsetVersion(12)
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:3125]: ONNX Celu only takes float inputs. How about pytorch? Have you tested this on other input data types?

Comment (KsenijaS) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:3146]: I think alpha is float in pytorch too https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1251
I added one test where alpha is int

Comment (neginraoof) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:3146]: No I'm not asking about alpha, I'm asking about inputs.

File: torch/onnx/symbolic_opset12.py [hunk 4]
```diff
@@ -60,5 +60,10 @@ def nll_loss2d(g, self, target, weight, reduction, ignore_index):
     return nll_loss(g, self, target, weight, reduction, ignore_index)
 
 
+def celu(g, self, alpha):
+    alpha = sym_help._maybe_get_const(alpha, 'f')
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset12.py:64]: What if self is not float?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 3 above)
Comment (neginraoof) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:3146]: You can create a test for an int input using torch.randint()

File: torch/onnx/symbolic_opset12.py (hunk 4 above)
Comment (KsenijaS) [reply torch/onnx/symbolic_opset12.py:64]: is Cast node needed?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 3 above)
Comment (KsenijaS) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:3146]: I think pytorch CCELU only accepts foalt inputs, tried with using int inputs (torch.randint()) and got the following:
RuntimeError: "elu_cpu" not implemented for 'Long'

File: torch/onnx/symbolic_opset12.py (hunk 4 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset12.py:64]: You mentioned int/long is not supported in PyTorch.
But looks like double is supported in PyTorch. However, ONNX Celu does not support it.
So we need a cast for double inputs.
Pleas verify this for all input types in your tests, and add casts where needed.

File: test/onnx/test_utility_funs.py [hunk 5]
```diff
@@ -591,6 +591,34 @@ def forward(self, x):
 
         np.testing.assert_allclose(ratio_pytorch, ratio_ort, rtol=0.01, atol=0.01)
 
+    @skipIfUnsupportedMinOpsetVersion(12)
```
Comment (neginraoof) [inline test/onnx/test_utility_funs.py:594]: Maybe we can remove or replace this test. I think the test for float64 in onnxruntime tests in good enough and covers casting. But if you want to have another test, maybe put it in test_operators. Thanks.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks.

**Comment by KsenijaS** [comment]
@houseroad please review

**Comment by KsenijaS** [comment]
@houseroad for review. Thanks!

**Review by ezyang — APPROVED** [review]
@KsenijaS please go ahead and land this

**Comment by KsenijaS** [comment]
@ezyang I don't think I have privileges to land a PR.

**Comment by KsenijaS** [comment]
@ezyang can you please merge this PR. Thanks!

**Comment by ezyang** [comment]
Oh, @houseroad imported it, I'll let him handle it

**Comment by KsenijaS** [comment]
@houseroad can you please merge this PR. Thanks!

**Comment by houseroad** [comment]
Sure, will land it soon.

**Pull request closed without merging**